lib/classes/PluginManager.js

```diff
@@ -90,17 +90,23 @@ class PluginManager {
 
         this.addPlugin(Plugin);
       } catch (error) {
-        // Rethrow the original error in case we're in debug mode.
-        if (process.env.SLS_DEBUG) {
-          throw error;
+        let errorMessage;
+        if (error && error.code === 'MODULE_NOT_FOUND' && error.message.endsWith(`'${plugin}'`)) {
```

Comment:
Is the `endsWith` necessary here or would the `MODULE_NOT_FOUND` be enough to check for the error?

The `endsWith` might not be stable across all supported Node versions.

Reply:
@HyperBrain `endsWith` is supported in Node v4, but not in Node v0.12. Does Serverless support Node v0.12 for some reason? If so, I can rewrite it with `indexOf()`

It ensures that it's about this specific require and not nested require. e.g. plugin itself may be loaded but require call within it may crash, and I believe that should be qualified as _initialization_ error and not _plugin not found_ error.

Reply:
I do not mean `endsWith` itself, but the semantics you use here 😄 .

If you check the ending of the message and rely on it to be the plugin name, this behavior might break or be unstable across Node versions, because especially error messages are most likely not standardized, only error codes are. So Node X might return a slightly different string as Node Y and it will break.

Reply:
@HyperBrain yes it's sort of compromise. It's to have detection accurate, so we know that this very require failed and not nested one, but as you noted to have that we need to rely on not guaranteed message format.

Speaking from my experience it's way more likely to approach nested require error (detected properly with above case), than node changing this error message format (It didn't happen since early days of node).

Still even if this message on node.js side changes, there's no real harm done. It's just that not found error would be reported as initialization error, that's it.

Anyway if you're strong about changing it it's totally fine for me. Let me know.

Reply:
@medikoo Yes, you're right. If we'd remove the check, the nested errors (i.e. missing modules _IN_ the plugin) would still lead to a misleading error message. I agree that the nested case is much more likely, especially during plugin development.
So, finally, I'm in favor of your current solution 😃